extensions/optaplanner/deployment/src/main/java/io/quarkus/optaplanner/deployment/OptaPlannerProcessor.java

```diff
@@ -0,0 +1,124 @@
+
+    @BuildStep
+    CapabilityBuildItem capability() {
+        return new CapabilityBuildItem(Capabilities.OPTAPLANNER);
+    }
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.OPTAPLANNER);
+    }
+
+    @BuildStep
+    void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        // The bean encapsulating the SolverFactory
+        additionalBeans.produce(new AdditionalBeanBuildItem(SolverFactoryProvider.class));
+    }
+
+    @BuildStep
+    @Record(STATIC_INIT)
+    void recordSolverFactory(OptaPlannerRecorder recorder,
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<BeanContainerListenerBuildItem> beanContainerListener) {
+        IndexView indexView = combinedIndex.getIndex();
+        String solutionClassName = findSolutionClassName(indexView);
+        List<String> entityClassNameList = findEntityClassNameList(indexView);
+        String constraintProviderClassName = findConstraintProviderClassName(indexView);
+        beanContainerListener
+                .produce(new BeanContainerListenerBuildItem(
+                        recorder.initializeSolverFactory(solutionClassName, entityClassNameList, constraintProviderClassName)));
+    }
+
+    private String findSolutionClassName(IndexView indexView) {
+        Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(DotNames.PLANNING_SOLUTION);
+        if (annotationInstances.size() > 1) {
+            throw new IllegalStateException("Multiple classes (" + convertAnnotationInstancesToString(annotationInstances)
+                    + ") found with a @" + PlanningSolution.class.getSimpleName() + " annotation.");
+        }
+        if (annotationInstances.isEmpty()) {
+            throw new IllegalStateException("No classes (" + convertAnnotationInstancesToString(annotationInstances)
+                    + ") found with a @" + PlanningSolution.class.getSimpleName() + " annotation.");
+        }
+        AnnotationTarget solutionTarget = annotationInstances.iterator().next().target();
+        if (solutionTarget.kind() != AnnotationTarget.Kind.CLASS) {
+            throw new IllegalStateException("A target (" + solutionTarget
+                    + ") with a @" + PlanningSolution.class.getSimpleName() + " must be a class.");
+        }
+
+        return solutionTarget.asClass().name().toString();
+    }
+
+    private List<String> findEntityClassNameList(IndexView indexView) {
+        Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(DotNames.PLANNING_ENTITY);
+        if (annotationInstances.isEmpty()) {
+            throw new IllegalStateException("No classes (" + convertAnnotationInstancesToString(annotationInstances)
+                    + ") found with a @" + PlanningEntity.class.getSimpleName() + " annotation.");
+        }
+        List<AnnotationTarget> targetList = annotationInstances.stream()
```

Comment:
Although not critical (since this is just the build time module), in general the Quarkus codebase tries not use to lambdas.